pnet-idp-client-saml2/src/main/java/at/porscheinformatik/idp/saml2/Saml2Utils.java

```diff
@@ -108,7 +114,11 @@ public static UriComponentsBuilder setRelayState(UriComponentsBuilder uriCompone
 
     public static Optional<String> getRelayState(HttpServletRequest request)
     {
-        return Optional.ofNullable(request.getParameter(RELAY_STATE_PARAM));
+        return Optional.ofNullable(request.getParameter(RELAY_STATE_PARAM))
+            .map(rs -> {
+                Matcher matcher = AUTO_GENERATED_RELAY_STATE_PATTERN.matcher(rs);
+                return matcher.matches() ? matcher.group() : rs;
+            }).filter(rs -> !rs.isEmpty());
```

Comment:
Nice catch. 👍 Completely forgot about the filter method and used "map" instead 😆

Reply:
Thanks